### ISSUE TITLE
Signature for sendas

### ISSUE THREAD

**Issue by morrisghi** [kam193/thbrd-signature#4]
The plugin is perfect, but if on G Workspace the user has a sendas, so he has the possibility to write with another address and with another signature, it would be nice to be able to find a way to find the signature for the sendas as well. The ideal would be, through the tool, to connect to Workspace, find primary and secondary emails (or other emails) therefore all the signatures for all accounts found , and then choose through the plug-in which signature to associate with  sendas on Thunderbird. Example: 

User: Morris

morris@mycompany.com <primary email> (signature X) 

morris@myothercompany.com <secondary email> (signature Y) - on which, in G Workspace, I simply have a sendas with another signature -

Both domains are registered on the same google control panel.

**Comment by kam193** [comment]
Hi, I'll definitively look at this. I had the idea of handling multiple signatures in the head, but I've never needed them.

I think I can get all signatures from Gmail and attach them do different identities in Thunderbird - do you think it's an idea that could solve the issue?

**Comment by morrisghi** [comment]
In my opinion, yes, and it would give the ideal. It would solve many
problems. You know, during migrations, many users struggle to switch to
Gmail, or at least the change happens gradually. Managing multiple
signatures massively in Gmail and at the same time on Thunderbird
(connected to Gmail) is a really expensive job, especially when the
signatures change often and moreover they are nominative. So the tool
(managing multiple signatures) allows you to focus on a single point (which
is Gmail) and forget about the changes to be made even in Thunderbird. So
it would be hyper useful!

Obviously I remain available to test the changes and for other suggestions!

Thank you for your time!




Il mer 19 gen 2022, 14:05 Kamil Mańkowski ***@***.***> ha
scritto:

**Comment by kam193** [comment]
Hey, I've checked what I can do, and there is a first test version for syncing Thunderbird's identities & Gmail's sendAs. Could you please install it (in Thunderbird: Manage Add-ons/the gear icon/Install Add-On From File), and see if it works for the basic scenario? In Add-on preferences, you will see the identities and available aliases. Could you verify this is what we want?

Don't worry about some visible issues or so - it's a first version, just want to check if we're on the same page & things work :)

Here is the file, but you may need to change the extension to `.xpi` (do not extract, just change the extension - Gtihub blocked .xpi, even if it's just the ZIP archive).
[signature-extension.zip](https://github.com/kam193/thbrd-signature/files/7927613/signature-extension.zip)

**Comment by morrisghi** [comment]
Ok today I test it and inform you!

👍

Il lun 24 gen 2022, 18:42 Kamil Mańkowski ***@***.***> ha
scritto:

**Comment by morrisghi** [comment]
1) I opened Thunderbird v 91.5.0 and I removed the old Plugin.
2) I went to Account Settings and removed the signature that automatically
downloaded by the old plugin
3) I closed  Thunderbird  and reopened it
4) In the Plugin Settings, I have already found the drop-down menu of all
the SendAs that I had set on Gmail. So for each email set on Thunderbird I
can choose which signature to exit with. I didn't set the OAuth because I
believe it recognized the old permission given earlier.
Sounds great to me!
It may already be OK!



Il giorno mar 25 gen 2022 alle ore 07:26 Maurizio Caurio <
***@***.***> ha scritto:

**Comment by kam193** [comment]
Great! Did it sync signatures correctly?

I need to improve & clean up the code, so give me some time more, but I'm happy it's working :)

**Comment by morrisghi** [comment]
Yes,

even though on thunderbird I only have two emails that I can send from, the
plugin shows me all sendas I have on gmail correctly. It's perfect!

[image: image.png]

I also tried to change the signature from gmail, I closed and reopened
Thunderbird and it automatically fetched the change.

Great!!





Il giorno mar 25 gen 2022 alle ore 09:16 Kamil Mańkowski <
***@***.***> ha scritto:

**Comment by kam193** [comment]
Hey, the new version has just landed in the Add-on portal. Please test and let me know if everything is working! https://addons.thunderbird.net/en-US/thunderbird/addon/signature-sync-for-gmail/

**Comment by morrisghi** [comment]
Hi Kamil,

works fine!

surely in the next few days I will be able to test it, even on more than
one person. I will report any anomalies.

Thanks for your availability!

Maurizio

Il giorno mer 9 feb 2022 alle ore 17:34 Kamil Mańkowski <
***@***.***> ha scritto:

**Comment by kam193** [comment]
Great! I'm happy I could help you :)

PS. Feel free to rate add-on on the Thunderbird website if you wish ;)

**Comment by morrisghi** [comment]
Review done!

Kamil, I have another challenge to offer you. Still on Google's sendas.
I'll explain: obviously on Google I can have a person who has more sendas
that can also refer to secondary domains (registered on Google of course).
Example, I maurizio, I can send as ***@***.*** and
***@***.***

On thunderbird however, once ***@***.*** has been entered, I do
not have the possibility to select FROM: and I am forced to enter another
account ***@***.*** (IMAP) which allows me to select the FROM of
Thunderbird and decide from which "mailbox" to send ....

In your opinion, there is a way to enter a Google account in Thunderbird,
see if that account has multiple sendas, and have the sendas from which I
can sent appear in the FROM of Thunderbird.. It would save me from
registering multiple accounts for the same mailbox in Thunderbird, because
it's basically a single main account with many sendas ...



Il giorno gio 10 feb 2022 alle ore 11:21 Kamil Mańkowski <
***@***.***> ha scritto:

**Comment by kam193** [comment]
Hey, the GitHub censored email examples, so I'm uncertain if I fully understand. Doesn't your case just require using multiple _identities_? I mean, in Thunderbird in Account Settings, you have a 'Manage Identities' button (as on screenshot below). Here you can add new identity and use any email address you wish - Thunderbird accepts anything, the mail server may reject incorrect during sending emails. Once you add a new identity, it's available in the FROM field.

![obraz](https://user-images.githubusercontent.com/8472271/153420604-40553c67-e7d7-4360-be9f-15d3642f63cb.png)

**Comment by morrisghi** [comment]
Yes, in fact at the moment it makes little sense. Maybe I'll find something
interesting to develop! :P

Thanks so much!

Il giorno gio 10 feb 2022 alle ore 17:33 Kamil Mańkowski <
***@***.***> ha scritto:

**Comment by kam193** [comment]
No problem, I'm happy that my program helps not only me ;)
